lib/srv6.h

```diff
@@ -167,12 +175,21 @@ static inline bool sid_diff(
 	return !sid_same(a, b);
 }
 
-static inline bool sid_zero(
-		const struct in6_addr *a)
+
+static inline bool sid_zero(const struct seg6_seg_stack *a)
+{
+	struct in6_addr zero = {};
+
+	if (!a)
+		return true;
+	return sid_same(&a->seg[0], &zero);
+}
+
+static inline bool sid_zero_ipv6(const struct in6_addr *a)
 {
 	struct in6_addr zero = {};
 
-	return sid_same(a, &zero);
```

Comment:
To further the point here as well about returning true when NULL.  Programmers are going to expect the the interfaces to very similiar functions are going to be the same, but here it is not.  Why?

Reply:
Yes, as mentioned in the discussion below, I addressed a function "sid_zero()". the "TRUE" return when pointer is NULL is replaced with assert. The logic to verify if pointer outside of the "sid_zero()".  Is that okay for you?